server/fleet/setup_experience.go

```diff
@@ -0,0 +1,47 @@
+package fleet
+
+type SetupExperienceStatusResultStatus string
+
+const (
+	SetupExperienceStatusPending SetupExperienceStatusResultStatus = "pending"
+	SetupExperienceStatusRunning SetupExperienceStatusResultStatus = "running"
+	SetupExperienceStatusSuccess SetupExperienceStatusResultStatus = "success"
+	SetupExperienceStatusFailure SetupExperienceStatusResultStatus = "failure"
+)
+
+// SetupExperienceStatusResult represents the status of a particular step in the macOS setup
+// experience process for a particular host. These steps can either be a software installer
+// installation, a VPP app installation, or a script execution.
+type SetupExperienceStatusResult struct {
+	ID                      uint                              `db:"id" json:"-" `
+	HostUUID                string                            `db:"host_uuid" json:"-" `
+	Name                    string                            `db:"name" json:"name,omitempty" `
+	Status                  SetupExperienceStatusResultStatus `db:"status" json:"status,omitempty" `
+	SoftwareInstallerID     *uint                             `db:"software_installer_id" json:"-" `
```

Comment:
This only returns ids but that's because the idea is that the swift dialog will then request the details of those software installers/scripts from the server to fill the UI, right?

Reply:
@mna I believe so. Worst case we can update this here with additional fields as needed.